pkg/odo/cli/component/create.go

```diff
@@ -923,7 +924,7 @@ func (co *CreateOptions) Run() (err error) {
 						},
 						Filepath: DevfilePath,
 					}
-					err = util.DownloadFile(params)
+					err = util.DownloadFile(params, 0)
```

Comment:
can we make this `0` a var-args or a functional parameter or make a new function called `DownloadFileWithCache(a,b)` which takes the second param because at many places its being set to `0` so would be better that the standard function has `DownloadFile` ( this wouldn't have the `cacheFor` param ) and a function `DownloadFileWithCache` that takes the `cacheFor`.

Reply:
Wouldn't that just complicate things?

Reply:
@girishramnani someting like this 21d5f510aec836ad5f2cc395864cb097e25bf2bf?

Reply:
> Wouldn't that just complicate things?

Would complicate the code but make the interface cleaner for whoever uses it